## 异常
![error和exception](捕获.PNG)
### 抛出异常
> 当方法中发生错误时，该方法会：①创建一个Exception对象。②接下来，该方法将异常传递给 JVM 运行时系统。

### 在 Java 中有两种不同类型的异常：
> catch已检查的异常将在编译时被捕获，并且在它们处于块中或被抛出之前不允许构建代码 。编译器不检查未检查（或运行时）异常。
1. 未经检查的异常
- 未经检查的异常是编译器未知的异常。
- 因为这些异常只有在运行时才知道，所以也称为运行时异常。
- 它们是编程错误的结果，通常是算术错误（例如被 0 除）。
- 当我们预计方法的调用者无法从异常中恢复时，使用未经检查的异常。
2. 检查异常
- 编译器知道已检查的异常。
- 如果我们正在调用一个可能会抛出已检查异常的方法，则必须对其进行处理（否则我们会从编译器中得到一个错误）。
- 当我们期望方法的调用者可以从异常中恢复时，使用已检查的异常。
```java
try {
    read();
}
catch (FileNotFoundException ex){
    ex.getLocalizedMessage();
}
finally {
}
```
![异常处理过程](捕获2.PNG)
## 枚举

## Dates 和 Calendar

## 正则表达式 (RegEx)

## 高级String